fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java

```diff
@@ -1341,9 +1341,13 @@ private void handleQueryStmt() throws Exception {
 
         // Sql and PartitionCache
         CacheAnalyzer cacheAnalyzer = new CacheAnalyzer(context, parsedStmt, planner);
-        if (cacheAnalyzer.enableCache() && !isOutfileQuery && queryStmt instanceof SelectStmt) {
-            handleCacheStmt(cacheAnalyzer, channel, (SelectStmt) queryStmt);
-            return;
+        if (cacheAnalyzer.enableCache() && !isOutfileQuery
+                && context.getSessionVariable().getSqlSelectLimit() == 9223372036854775807L
```

Comment:
change the behavior of `getSqlSelectLimit`, if `sqlSelectLimit` < 0 || `sqlSelectLimit` >= LONG.MAX_VALUE then return -1;
and then update this if statement to `context.getSessionVariable().getSqlSelectLimit() < 0`
do same thing to `getDefaultOrderByLimit`
and let sqlSelectLimit and defaultOrderByLimit as private.

Reply:
done